diff --git a/dist/esm/components/ReadOnlyFunctionForm.js b/dist/esm/components/ReadOnlyFunctionForm.js
index fbd52a25ae0e549215ec0e0d9392f99671f3260c..29f7ee506a0ba0a66f7f312d9dc9f8fcb64cf465 100644
--- a/dist/esm/components/ReadOnlyFunctionForm.js
+++ b/dist/esm/components/ReadOnlyFunctionForm.js
@@ -39,7 +39,7 @@ export const ReadOnlyFunctionForm = ({ contractAddress, abiFunction, inheritedFr
                 setForm(updatedFormValue);
             }, form: form, stateObjectKey: key, paramType: input }, key));
     });
-    return (_jsxs("div", { className: "flex flex-col gap-3 py-5 first:pt-0 last:pb-1", children: [_jsxs("p", { className: "font-medium my-0 break-words", children: [abiFunction.name, _jsx(InheritanceTooltip, { inheritedFrom: inheritedFrom })] }), inputElements, _jsxs("div", { className: "flex flex-col md:flex-row justify-between gap-2 flex-wrap", children: [_jsx("div", { className: "grow w-full md:max-w-[80%]", children: result !== null && result !== undefined && (_jsxs("div", { className: "bg-sui-primary-subtle text-sm px-4 py-1.5 break-words overflow-auto", children: [_jsx("p", { className: "font-bold m-0 mb-1", children: "Result:" }), _jsx("pre", { className: "whitespace-pre-wrap break-words", children: displayTxResult(result, "sm") })] })) }), _jsxs("button", { className: "btn-dc btn-dc-secondary btn-sm self-end md:self-start", onClick: async () => {
+    return (_jsxs("div", { className: "flex flex-col gap-3 py-5 first:pt-0 last:pb-1", children: [_jsxs("div", { className: "font-medium my-0 break-words flex items-center gap-1.5", children: [abiFunction.name, _jsx(InheritanceTooltip, { inheritedFrom: inheritedFrom })] }), inputElements, _jsxs("div", { className: "flex flex-col md:flex-row justify-between gap-2 flex-wrap", children: [_jsx("div", { className: "grow w-full md:max-w-[80%]", children: result !== null && result !== undefined && (_jsxs("div", { className: "bg-sui-primary-subtle text-sm px-4 py-1.5 break-words overflow-auto", children: [_jsx("p", { className: "font-bold m-0 mb-1", children: "Result:" }), _jsx("pre", { className: "whitespace-pre-wrap break-words", children: displayTxResult(result, "sm") })] })) }), _jsxs("button", { className: "btn-dc btn-dc-secondary btn-sm", onClick: async () => {
                             const { data } = await refetch();
                             setResult(data);
                         }, disabled: isFetching, children: [isFetching && _jsx("span", { className: "loading-dc loading-dc-spinner loading-xs" }), "Read \uD83D\uDCE1"] })] })] }));
diff --git a/dist/esm/components/Tooltip.js b/dist/esm/components/Tooltip.js
index 303af5f58cba465aa0501ad4edd4e9304b9d70bb..fe0af5fef99b1beedd1aadd82c862264fdce8f89 100644
--- a/dist/esm/components/Tooltip.js
+++ b/dist/esm/components/Tooltip.js
@@ -33,5 +33,5 @@ export const Tooltip = ({ children, content, position = "top", show = true, clas
     }
     const positionClasses = getPositionClasses(position);
     const arrowClasses = getArrowClasses(position);
-    return (_jsxs("div", { className: `relative group inline-block ${className}`, children: [children, _jsxs("div", { className: `absolute ${positionClasses} px-2 py-1 text-sm whitespace-nowrap opacity-0 group-hover:opacity-100 transition-opacity duration-200 pointer-events-none z-10 bg-sui-primary text-sui-primary-content ${contentClassName}`, children: [content, _jsx("div", { className: arrowClasses })] })] }));
+    return (_jsxs("span", { className: `relative group inline-block ${className}`, children: [children, _jsxs("span", { className: `absolute ${positionClasses} px-2 py-1 text-sm whitespace-nowrap opacity-0 group-hover:opacity-100 transition-opacity duration-200 pointer-events-none z-10 bg-sui-primary text-sui-primary-content ${contentClassName}`, children: [content, _jsx("span", { className: arrowClasses })] })] }));
 };
diff --git a/dist/esm/components/WriteOnlyFunctionForm.js b/dist/esm/components/WriteOnlyFunctionForm.js
index 57267c0d44c828f8623f6234bef576c57d51a0ad..47f997d928de4be12a9667286fba960b01d33b9a 100644
--- a/dist/esm/components/WriteOnlyFunctionForm.js
+++ b/dist/esm/components/WriteOnlyFunctionForm.js
@@ -56,7 +56,7 @@ export const WriteOnlyFunctionForm = ({ abi, abiFunction, onChange, contractAddr
             }, form: form, stateObjectKey: key, paramType: input }, key));
     });
     const zeroInputs = inputs.length === 0 && abiFunction.stateMutability !== "payable";
-    return (_jsxs("div", { className: "py-5 space-y-3 first:pt-0 last:pb-1", children: [_jsxs("div", { className: `flex gap-3 ${zeroInputs ? "flex-row justify-between items-center" : "flex-col"}`, children: [_jsxs("p", { className: "font-medium my-0 break-words", children: [abiFunction.name, _jsx(InheritanceTooltip, { inheritedFrom: inheritedFrom })] }), inputs, abiFunction.stateMutability === "payable" ? (_jsxs("div", { className: "flex flex-col gap-1.5 w-full", children: [_jsxs("div", { className: "flex items-center ml-2", children: [_jsx("span", { className: "text-xs font-medium mr-2 leading-none", children: "payable value" }), _jsx("span", { className: "block text-xs font-extralight leading-none", children: "wei" })] }), _jsx(IntegerInput, { value: txValue, onChange: (updatedTxValue) => {
+    return (_jsxs("div", { className: "py-5 space-y-3 first:pt-0 last:pb-1", children: [_jsxs("div", { className: `flex gap-3 ${zeroInputs ? "flex-row justify-between items-center" : "flex-col"}`, children: [_jsxs("div", { className: "font-medium my-0 break-words flex items-center gap-1.5", children: [abiFunction.name, _jsx(InheritanceTooltip, { inheritedFrom: inheritedFrom })] }), inputs, abiFunction.stateMutability === "payable" ? (_jsxs("div", { className: "flex flex-col gap-1.5 w-full", children: [_jsxs("div", { className: "flex items-center ml-2", children: [_jsx("span", { className: "text-xs font-medium mr-2 leading-none", children: "payable value" }), _jsx("span", { className: "block text-xs font-extralight leading-none", children: "wei" })] }), _jsx(IntegerInput, { value: txValue, onChange: (updatedTxValue) => {
                                     setDisplayedTxResult(undefined);
                                     setTxValue(updatedTxValue);
                                 }, placeholder: "value (wei)" })] })) : null, _jsxs("div", { className: "flex justify-between gap-2 min-w-0", children: [!zeroInputs && (_jsx("div", { className: "flex-1 min-w-0 max-w-11/12 overflow-auto", children: displayedTxResult ? _jsx(TxReceipt, { txResult: displayedTxResult }) : null })), _jsx("div", { className: "flex flex-shrink-0 self-start", children: _jsx(Tooltip, { content: "Wallet not connected or in the wrong network", position: "bottom", show: writeDisabled, children: _jsxs("button", { className: "btn-dc btn-dc-secondary btn-sm", disabled: writeDisabled || isPending, onClick: handleWrite, children: [isPending && _jsx("span", { className: "loading-dc loading-dc-spinner mr-1" }), "Send \uD83D\uDCB8"] }) }) })] })] }), zeroInputs && txResult ? (_jsx("div", { className: "w-full min-w-0 overflow-auto", children: _jsx(TxReceipt, { txResult: txResult }) })) : null] }));
